source/val/validate_cfg.cpp

```diff
@@ -330,7 +330,7 @@ spv_result_t StructuredControlFlowChecks(
     uint32_t header_block;
     tie(back_edge_block, header_block) = back_edge;
     if (!function->IsBlockType(header_block, kBlockTypeLoop)) {
-      return _.diag(SPV_ERROR_INVALID_CFG)
+      return _.diag(SPV_ERROR_INVALID_CFG, _.FindDef(back_edge_block))
```

Comment:
I guessed at these three, but they seem to make sense (and give a better disassembly then end of file).